packages/atst/src/cli.ts

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env node
+import { cac } from 'cac'
+import type { Address } from '@wagmi/core'
+
+import { readOptionsValidators, ReadOptions } from './commands/read'
+import * as logger from './lib/logger'
+// @ts-ignore it's mad about me importing something not in tsconfig.includes
+import packageJson from '../package.json'
+import { writeOptionsValidators } from './commands/write'
+
+const cli = cac('atst')
+
+cli
+  .command('read', 'read an attestation')
+  .option('--creator <string>', readOptionsValidators.creator.description!)
+  .option('--about <string>', readOptionsValidators.about.description!)
+  .option('--key <string>', readOptionsValidators.key.description!)
+  .option('--data-type [string]', readOptionsValidators.dataType.description!, {
+    default: readOptionsValidators.dataType.parse(undefined),
+  })
+  .option('--rpc-url [url]', readOptionsValidators.rpcUrl.description!, {
+    default: readOptionsValidators.rpcUrl.parse(undefined),
+  })
+  .option('--contract [address]', readOptionsValidators.contract.description!, {
+    default: readOptionsValidators.contract.parse(undefined),
+  })
+  .example(
+    () =>
+      `atst read --key "optimist.base-uri" --about 0x2335022c740d17c2837f9C884Bfe4fFdbf0A95D5 --creator 0x60c5C9c98bcBd0b0F2fD89B24c16e533BaA8CdA3`
+  )
+  .action(async (options: ReadOptions) => {
+    const { read } = await import('./commands/read')
+
+    // TODO use the native api to do this instead of parsing the raw args
+    // by default options parses addresses as numbers without precision
+    // we should use the args parsing library to do this directly
+    // but for now I didn't bother to figure out how to do that
```

Comment:
```suggestion
```
Could remove comment no chance this ever actually gets prioritized if we don't do it now